### PR TITLE
Issue #1782: Upgraded/added dependencies needed for Java 9/10 support.

### LINKED ISSUE #1782: DataCleaner doesn't start on Java 10
Just installed Java 10, and now I'm unable to run DataCleaner. Here's the exception that I see in the log file:

```
INFO  15:55:09.602 [main] Bootstrap - Welcome to DataCleaner 5.4.0
INFO  15:55:09.602 [main] Bootstrap - CLI mode=false, use -usage to view usage options
INFO  15:55:09.930 [main] LookAndFeelManager - Look and feel set to: [The JGoodies Plastic XP Look and Feel - © 2001-2009 JGoodies Karsten Lentzsch - com.jgoodies.looks.plastic.PlasticXPLookAndFeel]
WARN  15:55:10.713 [main] LazyRef - Failed to fetch value: org.datacleaner.user.DataCleanerConfigurationReader@fac80. Reporting error.
java.lang.NoClassDefFoundError: javax/xml/bind/JAXBException
	at org.datacleaner.user.DataCleanerConfigurationReader.fetch(DataCleanerConfigurationReader.java:84)
	at org.datacleaner.user.DataCleanerConfigurationReader.fetch(DataCleanerConfigurationReader.java:48)
	at org.apache.metamodel.util.LazyRef.get(LazyRef.java:47)
	at org.datacleaner.guice.DCModuleImpl.getUndecoratedAnalyzerBeansConfiguration(DCModuleImpl.java:248)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:104)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
	at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
	at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
	at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:89)
	at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:98)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
	at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1024)
	at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1013)
	at org.datacleaner.bootstrap.Bootstrap.runInternal(Bootstrap.java:177)
	at org.datacleaner.bootstrap.Bootstrap.run(Bootstrap.java:92)
	at org.datacleaner.Main.main(Main.java:165)
	at org.datacleaner.Main.main(Main.java:150)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.JAXBException
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	... 24 more
ERROR 15:55:10.730 [main] Bootstrap - An unexpected error has occurred during bootstrap. Exiting with status code -2.
com.google.inject.ProvisionException: Guice provision errors:

1) Error in custom provider, java.lang.NullPointerException
  at org.datacleaner.guice.DCModuleImpl.getDataCleanerConfiguration(DCModuleImpl.java:254)
  while locating org.datacleaner.configuration.DataCleanerConfiguration

1 error
	at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:987)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1013)
	at org.datacleaner.bootstrap.Bootstrap.runInternal(Bootstrap.java:177)
	at org.datacleaner.bootstrap.Bootstrap.run(Bootstrap.java:92)
	at org.datacleaner.Main.main(Main.java:165)
	at org.datacleaner.Main.main(Main.java:150)
Caused by: java.lang.NullPointerException
	at org.datacleaner.guice.DCModuleImpl.getDataCleanerConfiguration(DCModuleImpl.java:260)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:104)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
	at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1024)
	at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
	... 5 more
INFO  15:55:10.738 [main] DCExitActionListener - Scheduling shutdown thread
```

I guess it has to do with JAXB and module loading etc.

### PR DESCRIPTION
**Pull request by kaspersorensen**
Fixes #1782

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -784,6 +786,26 @@
...
+			</dependency>
+			<dependency>
+				<groupId>javax.activation</groupId>
+				<artifactId>activation</artifactId>
+				<version>1.1.1</version>
+			</dependency>
+			<dependency>
+				<groupId>javax.xml.ws</groupId>
+				<artifactId>jaxws-api</artifactId>
+				<version>2.3.0</version>
```
Comment (LosD) [inline pom.xml]: Shouldn't these all use a version variable?

**Comment by LosD** [comment]
Not much to say, code-wise :)

So LGTM, except (maybe, if it even matters anymore) not using our version variables.

File: pom.xml (hunk 1 above)
Comment (kaspersorensen) [reply pom.xml]: I don't know if it matters when the version is only referenced once. At least we seem to have quite a few other dependencies defined without a property variable. I think the properties are mostly useful for something like Spring or MetaModel or whatever, where the same version if referenced again and again.

Comment (LosD) [reply pom.xml]: Fine with me, then. In general though, I'd prefer we either do it consistently or not at all. When it's there, you're kinda led to think it's the go-to place to look for versions. :)

In the long term, a complete parent POM dependencyManagment section along with some evil build-breaking checker for other places _might_ be better. That brings its own problems, though.

**Comment by LosD** [comment]
Updated comments, but you're free to just ignore me. I kinda haven't touched this in ages, so my insights are definitely waning.

**Comment by kaspersorensen** [comment]
I'm gonna merge it then :-) I think it's fine for someone who cares to externalize all version numbers to properties, but I'm not going to be that person myself ;-) I guess I see the point, just don't see it enough to go through that process myself :-P

**Comment by LosD** [comment]
:D

**Pull request merged**